staging/src/k8s.io/apiserver/pkg/storage/selection_predicate_test.go

```diff
@@ -123,3 +128,62 @@ func TestSelectionPredicate(t *testing.T) {
 		}
 	}
 }
+
+func TestSelectionPredicateMatcherIndex(t *testing.T) {
+	testCases := map[string]struct {
+		labelSelector, fieldSelector string
+		indexLabels                  []string
+		indexFields                  []string
+		excepted                     []MatchValue
+	}{
+		"Match nil": {
+			labelSelector: "name=foo",
+			fieldSelector: "uid=12345",
+			indexLabels:   []string{"bar"},
+			indexFields:   []string{},
+			excepted:      nil,
+		},
+		"Match field": {
+			labelSelector: "name=foo",
+			fieldSelector: "uid=12345",
+			indexLabels:   []string{},
+			indexFields:   []string{"uid"},
+			excepted:      []MatchValue{{IndexName: FieldIndex("uid"), Value: "12345"}},
+		},
+		"Match label": {
+			labelSelector: "name=foo",
+			fieldSelector: "uid=12345",
+			indexLabels:   []string{"name"},
+			indexFields:   []string{},
+			excepted:      []MatchValue{{IndexName: LabelIndex("name"), Value: "foo"}},
+		},
+		"Match field and label": {
+			labelSelector: "name=foo",
+			fieldSelector: "uid=12345",
+			indexLabels:   []string{"name"},
+			indexFields:   []string{"uid"},
+			excepted:      []MatchValue{{IndexName: FieldIndex("uid"), Value: "12345"}, {IndexName: LabelIndex("name"), Value: "foo"}},
+		},
+	}
```

Comment:
add a test case for a field selector and label selector that specify negative matches on index keys and ensure MatcherIndex does not include them

```
		"Negative match field and label": {
			labelSelector: "name!=foo",
			fieldSelector: "uid!=12345",
			indexLabels:   []string{"name"},
			indexFields:   []string{"uid"},
			expected:      nil,
		},
```

Reply:
> add a test case for a field selector and label selector that specify negative matches on index keys and ensure MatcherIndex does not include them
> 
> ```
> 		"Negative match field and label": {
> 			labelSelector: "name!=foo",
> 			fieldSelector: "uid!=12345",
> 			indexLabels:   []string{"name"},
> 			indexFields:   []string{"uid"},
> 			expected:      nil,
> 		},
> ```

done